core/modules/parsers/wikiparser/wikiparser.js

```diff
@@ -165,6 +165,21 @@ WikiParser.prototype.findNextMatch = function(rules,startPos) {
 	return matchingRule;
 };
 
+WikiParser.prototype.parseRule = function(rule) {
+	var start = this.pos,
+		blocks = rule.parse();
+	for(var i=0; i<blocks.length; i++) {
+		var block = blocks[i];
+		if(block.start !== undefined) {
```

Comment:
what is the extra `if ()` element needed for? .. If there are no blocks, imo the for shouldn't be executed at all. ... It should be a "return early" approach.

Reply:
In the ideal case every block would define its own range and this for loop wouldn't be needed at all.

In the real world some blocks define their range and some don't, so this is a fallback for child blocks that don't already have start/end ranges defined (and are therefore `undefined`) - in this case it falls back to using the range of the parent.

I'll add a comment in code to make this clearer.

Reply:
I realize now that it's the opposite of what I intended 🤦‍♂️